toolkit/scripts/rpmops.sh

```diff
@@ -19,15 +19,70 @@ DISTRO_MACRO="$(make -s -f $REPO_ROOT/toolkit/Makefile printvar-DIST_VERSION_MAC
 
 SPECS_DIR="$REPO_ROOT/SPECS"
+rpm_package_macros_file_path=""
+
+# Check if the MACROS_FILE_PATH is already defined, and the directory exists
+if [[ -z "$RPM_OPS_MACROS_DIRECTORY" ]] || [[ ! -d "$RPM_OPS_MACROS_DIRECTORY" ]]
+then
+    # Create a temporary directory to store the macros file, we don't want to use ./build since that is likely owned by root, but we also
+    # want to re-use the same macro files if possible to avoid downloads and handle multiple concurrent versions.
+    RPM_OPS_MACROS_DIRECTORY="$(mktemp -d)"
+    export RPM_OPS_MACROS_DIRECTORY
+fi
+
+# We do not want to use the host's default macros. Scan the toolchain manifest for rpm-libs and use it instead.
+# If this fails use the built-in macros from the host. This is the fallback option if the above generate an error.
+# Note: There is a limitation. In the event that the rpm-libs package itself is updated, the macros file will still
+# be the old one.
+arch="$(make -s -f $REPO_ROOT/toolkit/Makefile printvar-build_arch)"
+if [[ -f "$REPO_ROOT/toolkit/resources/manifests/package/toolchain_$arch.txt" ]]
+then
+    MACROS_PACKAGE_NAME=$(grep "rpm-libs" "$REPO_ROOT/toolkit/resources/manifests/package/toolchain_$arch.txt") || MACROS_PACKAGE_NAME=""
+fi
+
+if [[ -n $MACROS_PACKAGE_NAME ]]
+then
+    source_url=( $(make -s -f $REPO_ROOT/toolkit/Makefile printvar-PACKAGE_URL_LIST) )
+    # We assume the 1st entry will contain the correct URL
+    source_url_full=${source_url[0]}/$MACROS_PACKAGE_NAME
+    # To try and disambiguate different versions of the macros files, hash the url.
+    url_hash=$(echo $source_url_full | sha256sum | awk '{print $1}')
+    temp_macros_file_path="$RPM_OPS_MACROS_DIRECTORY/$url_hash"
+
+    # Check if we have the file already extracted
+    if [[ ! -f "$temp_macros_file_path" ]]
+    then
+        echo "Downloading and extracting macros file from package '$source_url_full' to '$temp_macros_file_path'" >&2
+        # Extract the macros file from the package into the macros file path.
+        # Use a temporary file to avoid partial writes if the download/extract fails.
+        curl -s "$source_url_full" | rpm2cpio - | cpio --quiet -i --to-stdout ./usr/lib/rpm/macros > "$temp_macros_file_path".tmp && \
+            mv "$temp_macros_file_path".tmp "$temp_macros_file_path" || \
+            temp_macros_file_path=""
+        if [[ -z "$temp_macros_file_path" ]]
+        then
+            echo "Warning: Failed to extract macros file from package $MACROS_PACKAGE_NAME", using built-in defaults >&2
+        fi
+
+    fi
+    rpm_package_macros_file_path="$temp_macros_file_path"
+fi
 
 # Azure Linux macro files used during spec parsing.
 MACROS=()
-for macro_file in "$SPECS_DIR"/azurelinux-rpm-macros/macros* "$SPECS_DIR"/pyproject-rpm-macros/macros.pyproject "$SPECS_DIR"/perl/macros.perl
+# If we have a package macros file, we will use it for parsing the spec files instead of the built-in macros
+if [[ -n $rpm_package_macros_file_path ]]
```

Comment:
Nit-pick to be more strict and clear about our expectations for `rpm_package_macros_file_path`:
```suggestion
if [[ -f $rpm_package_macros_file_path ]]
```